### DISCUSSION TITLE
commons-jxpath CVE-2022-41852 vulnerabiltiy

### DISCUSSION THREAD

**Discussion by kekolamb** [eclipse-pde/eclipse.pde#1220] [Q&A]
We are currently utilizing org.eclipse.pde.runtime version 3.7.300, which has a compile dependency on commons-jxpath. Unfortunately, commons-jxpath is vulnerable to CVE-2022-41852. Although jxpath has rejected this vulnerability, we are still exposed due to our current dependency. We are using Eclipse version 4.23.0. Is there a solution to mitigate this vulnerability for the eclipse version 4.23.0?

**Comment by merks** [comment]
In the IDE's use of this library, the xpaths being processed are paths into the local e4xmi model files, so they are not arbitrary paths that could potentially be from entrusted sources.

**Comment by akurtakov** [comment]
This is not the first time this topic comes up . See https://github.com/eclipse-platform/eclipse.platform.ui/issues/423 https://github.com/eclipse-platform/eclipse.platform/issues/291
